src/safeposix/syscalls/sys_calls.rs

```diff
@@ -370,41 +370,131 @@ impl Cage {
+    /// login time. The group id is the group of the user who invoked the program. 
+    /// Lind is only run in one group - and hence a default value is expected from this function.  
+    /// Initially we check if the call takes place during the loading stage, and return -1 if yes and set the 
+    /// gid to be the default value.
+    /// 
+    /// ### Arguments 
+    /// 
+    /// The `getgid_syscall` does not take any argument.
+    ///
+    /// ### Returns
+    /// 
+    /// Depending on whether the gid has been initialized or not this function returns either -1 
+    /// or the default gid as a 32 bit integer. 
     pub fn getgid_syscall(&self) -> i32 {
+        // The call returns -1 to indicate that the call happened during the loading stage
         if self.getgid.load(interface::RustAtomicOrdering::Relaxed) == -1 {
             self.getgid
                 .store(DEFAULT_GID as i32, interface::RustAtomicOrdering::Relaxed);
             return -1;
         }
         DEFAULT_GID as i32 //Lind is only run in one group so a default value is returned
-    }
+    } 
+
+    /// ### Description 
+    /// 
+    /// The `getegid_syscall` returns the effective group id of the user who invoked the process.
+    /// Since Lind is only run in one group a default value (or -1) is returned. 
+    /// Initially we check if the call takes place during the loading stage, and return -1 if yes and set the 
+    /// egid to be the default value.
+    /// 
+    /// ### Arguments 
+    /// 
+    /// This syscall does not take any arguments
+    /// 
+    /// ### Returns
+    /// 
+    /// Returns a 32 bit integer value (or -1) which represents the effective group
     pub fn getegid_syscall(&self) -> i32 {
+        // The call returns -1 to indicate that the call happened during the loading stage
         if self.getegid.load(interface::RustAtomicOrdering::Relaxed) == -1 {
             self.getegid
                 .store(DEFAULT_GID as i32, interface::RustAtomicOrdering::Relaxed);
             return -1;
         }
         DEFAULT_GID as i32 //Lind is only run in one group so a default value is returned
     }
-
+ 
+
+    /// ### Description
+    /// 
+    /// The `getuid_syscall` returns the real user id of the calling process. 
+    /// The real user id is the user who invoked the calling process. 
+    /// As Lind only allows one user, a default value is returned. 
+    /// Initially we check if the call takes place during the loading stage, and return -1 if yes and set the 
+    /// uid to be the default value.
+    /// 
+    /// ### Arguments
+    ///  
+    /// The `getuid_syscall` does not take any arguments
+    /// 
+    /// ### Returns
+    /// 
+    /// Returns a 32 bit default integer (or -1) representing the user
     pub fn getuid_syscall(&self) -> i32 {
+        // The call returns -1 to indicate that the call happened during the loading stage
```

Comment:
I think a more precise comment here would be:

"We return -1 for the first call for compatibility with the dynamic loader. For subsequent calls we return our default value."